### ISSUE TITLE
Reminder Lists Repeating

### ISSUE THREAD

**Issue by Bluechroma** [davidlday/RemindersWidget#17]
Hi, I'm not sure what I may be doing wrong, a newbie here. I have one list (To Do List) that repeats three times and other lists show up that I don't even have any more. Is there a cache for Reminders maybe? If I hide the offending list is hides all three copies. I've tried removing and adding them back, also syncing and unsyncing iCloud but with no changes.
Thanks for any help.

Catalina 10.15.7
iMac late 2012

**Comment by davidlday** [comment]
@Bluechroma - I don't know of anything off the top of my head that would cause the problem. Would you mind providing the following, please?

* Screenshot, so I can visualize the problem better
* In a terminal, please run `ls -al $HOME/Library/Reminders/Container_v1/Stores` and paste the output here

I'll probably have more questions, so please bear with me.

Thanks!

**Comment by davidlday** [comment]
@Bluechroma - I'm not seeing the screenshots. If you sent them as an attachment in your email, they didn't show up. Can you try uploading them directly on GitHub, pls? Thank you!

**Comment by Bluechroma** [comment]
![Screen Shot 2021-01-05 at 2 11 02 PM](https://user-images.githubusercontent.com/76921375/103858582-a04a5d80-5086-11eb-9a37-e2c52132b905.png)
![Screen Shot 2021-01-05 at 2 13 54 PM](https://user-images.githubusercontent.com/76921375/103858614-a9d3c580-5086-11eb-85df-41b982ac7ec3.png)

**Comment by davidlday** [comment]
One of your files above looks huge. Can you send me a couple more things, pls?

* A screenshot of Reminders
* Another listing of the Reminders directory using `ls -alh $HOME/Library/Reminders/Container_v1/Stores` (add the `h` flag)

Also, can you confirm again your OS? The dock in your top screenshot looks like Big Sur, not Catalina. Probably makes no difference, but just want to be sure.

**Comment by Bluechroma** [comment]
I'm running Catalina with custom icons on the dock, that's probably why it looks like Big Sur. 

here's a screenshot of Reminders
![Screen Shot 2021-01-08 at 5 01 41 PM](https://user-images.githubusercontent.com/76921375/104069813-93875000-51d3-11eb-89a8-da71cba5558a.png)

Last login: Thu Jan  7 20:14:26 on console
micky@Mickys-iMac ~ % ls -alh $HOME/Library/Reminders/Container_v1/Stores
total 801328
drwxr-xr-x  12 micky  staff   384B Jan  3 00:19 .
drwxr-xr-x   5 micky  staff   160B Feb  1  2020 ..
-rw-r--r--@  1 micky  staff   6.0K Jan  2 01:41 .DS_Store
-rw-r--r--@  1 micky  staff   383M Jan  8 16:27 Data-03F4949A-EEE7-49E8-ABF2-3DF791AA22DD.sqlite
-rw-r--r--@  1 micky  staff    32K Jan  8 02:05 Data-03F4949A-EEE7-49E8-ABF2-3DF791AA22DD.sqlite-shm
-rw-r--r--@  1 micky  staff   5.0M Jan  8 16:27 Data-03F4949A-EEE7-49E8-ABF2-3DF791AA22DD.sqlite-wal
-rw-r--r--   1 micky  staff   204K Jan  2 01:43 Data-06B30730-88FE-4C43-9438-93D360E8816E.sqlite
-rw-r--r--   1 micky  staff    32K Jan  8 02:05 Data-06B30730-88FE-4C43-9438-93D360E8816E.sqlite-shm
-rw-r--r--   1 micky  staff   769K Jan  3 13:28 Data-06B30730-88FE-4C43-9438-93D360E8816E.sqlite-wal
-rw-r--r--   1 micky  staff   204K Jan  2 01:43 Data-978C1BCF-9B38-44C6-81BC-8AC727C5F788.sqlite
-rw-r--r--   1 micky  staff    32K Jan  8 02:05 Data-978C1BCF-9B38-44C6-81BC-8AC727C5F788.sqlite-shm
-rw-r--r--   1 micky  staff   950K Jan  3 00:19 Data-978C1BCF-9B38-44C6-81BC-8AC727C5F788.sqlite-wal
micky@Mickys-iMac ~ %

**Comment by davidlday** [comment]
@Bluechroma - Sorry for not getting back to you sooner.

I'm honestly not sure what's going on here. Unfortunately everything about the todo database is undocumented. If you're willing to try some queries and post results, we might make some progress.

Let's start with:

```bash
sqlite3 "file:$HOME/Library/Reminders/Container_v1/Stores/Data-03F4949A-EEE7-49E8-ABF2-3DF791AA22DD.sqlite?mode=ro" "SELECT Z_ENT, ZNAME, ZNAME1 FROM ZREMCDOBJECT WHERE Z_ENT = 21;"
```
 You should get a list of all your lists, like: 

```bash
21||Work
21||Personal
21||Unprocessed
```

**Comment by Bluechroma** [comment]
I've been removing and adding some lists so that original one is no longer listed. It seems they only show when I sync to iCloud. I ran the command with the largest sqlite file listed and got this:

 micky@Mickys-iMac ~ % sqlite3 "file:$HOME/Library/Reminders/Container_v1/Stores/Data-31C2D823-75B2-4202-9496-E939AB26AA30.sqlite?mode=ro" "SELECT Z_ENT, ZNAME, ZNAME1 FROM ZREMCDOBJECT WHERE Z_ENT = 21;"
21||HHSC
21||System Tasks
21||Reminders
21||To Do List
21||Meds
21||To Do List
21||Daily Tasks
21||Meds
21||Reminders
21||Shopping List
21||My To Do List
21||Design Ideas
21||Shopping List
21||TO DO / THIS WEEKEND
21||Doris
21||House Grocery List
21||My Shopping List
21||My Shopping List
21||To Do
21||Daily Tasks
21||Doris
21||TO DO / THIS WEEKEND
21||To Do
21||Summer Camp 2020
21||House Grocery List
21||Summer Camp 2020
21||iPad Test
21||iMac Test Remindr
21||Personal Projects
21||🧠 BRAIN
21||test
21||Reminders
21||Shopping List
21||System Tasks
21||To Do List
21||Reminders
21||TEST-2-10
21||Test 2/10
21||Design Ideas
21||Reminders
21||Reminders
micky@Mickys-iMac ~ % 


Lots of repeating lists and old ones I've since deleted. I only have four lists now. "Shopping List", "My To Do List", "Design Ideas", and "Reminders."
Thanks again for the help!

**Comment by davidlday** [comment]
@Bluechroma  - Are you saying the widget is working for you now, or do you still need some help troubleshooting? Happy to help still if you need it.

**Comment by Bluechroma** [comment]
Yes and No. It's still not working for me right away, so I hid all the reminder lists that were problems and I only allowed a new list with a new name.  So its working for what I need it for but it seems like iCloud is not purging the older/previous lists. I'll need to figure out why these lists are not going away for me, so it doesn't seem like it's on your end. Thanks for the help though.

**Comment by davidlday** [comment]
Issue does not appear to be with the widget.

**Comment by davidlday** [comment]
This is an issue with the widget - was finally able to reproduce. Will have a fix out shortly.

**Comment by davidlday** [comment]
Lists and/or Tasks that have been deleted are not getting filtered out.

**Comment by davidlday** [comment]
Fixed via #19
